### ISSUE TITLE
spec.rules[0].http.paths[0].pathType: Unsupported value: "prefix": supported values: "Exact",  "ImplementationSpecific", "Prefix" on Minikube v1.23.0

### ISSUE THREAD

**Issue by OOPMan** [garden-io/garden#2573]
### Current Behavior

It seems like the version of K8S used by Minikube v1.23.0 does not like the `pathType` values being generated by Garden for Ingresses. Specifically, it expects the values to be propercase rather than allowing lowercase values. The version of K8S used by Minikube v1.23.0 is v1.22.1

This is not an issue on Minikube v1.21.0 which uses K8S v1.20.7. I have not tested on Minikube v1.22.0.

This causes deployments to fail with error output of the form:

```
Failed deploying service 'cim-datastore' (from module 'cim-datastore'). Here is the output:
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Command "/home/cgonsalves/.garden/tools/kubectl/11ac0bbd21058c50/kubectl --context=minikube 
--namespace=opusonesolutions-derms-default apply --prune --selector service=cim-datastore --output=json -f -" failed with code 1:

The Ingress "cim-datastore-0" is invalid: spec.rules[0].http.paths[0].pathType: Unsupported value: "prefix": supported values: 
"Exact", "ImplementationSpecific", "Prefix"

The Ingress "cim-datastore-0" is invalid: spec.rules[0].http.paths[0].pathType: Unsupported value: "prefix": supported values: 
"Exact", "ImplementationSpecific", "Prefix"
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

Failed deploying service 'keycloak' (from module 'keycloak'). Here is the output:
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Command "/home/cgonsalves/.garden/tools/kubectl/11ac0bbd21058c50/kubectl --context=minikube 
--namespace=opusonesolutions-derms-default apply --prune --selector service=keycloak --output=json -f -" failed with code 1:

The Ingress "keycloak-0" is invalid: spec.rules[0].http.paths[0].pathType: Unsupported value: "prefix": supported values: "Exact", 
"ImplementationSpecific", "Prefix"

The Ingress "keycloak-0" is invalid: spec.rules[0].http.paths[0].pathType: Unsupported value: "prefix": supported values: "Exact", 
"ImplementationSpecific", "Prefix"
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```

### Expected behavior

Expected services to be deployed without issue

### Reproducible example

It seems that any basic container service with an ingress declared will suffer this issue on K8S v1.22.1

### Workaround

Revert to Minikube v1.21.0 and/or start minikube as follows: `minikube start  --kubernetes-version=v1.20.7`

### Suggested solution(s)

Fix K8S manifest generation to propercase the value

### Your environment

* **OS:** Linux
* **How I'm running Kubernetes:** Minikube

Garden version `0.12.25`

**Comment by eysi09** [comment]
Thanks for the report, this should be a simple fix if using `Prefix` is backwards compatible.

**Comment by OOPMan** [comment]
> Thanks for the report, this should be a simple fix if using `Prefix` is backwards compatible. 

Hopefully it is. This is a bit of a weird one since the K8S docs all reference it via the propercase form but it seems until recently lowercase worked too and a cursory examination of their changelogs didn't turn up a reference to this change

**Comment by vvagaytsev** [comment]
@thsig is this still a valid issue? Should we reopen this?

**Comment by vvagaytsev** [comment]
Fixed in #2639
